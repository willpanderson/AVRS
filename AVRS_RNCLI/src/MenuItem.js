import { View, Text, Pressable, Image, StyleSheet } from 'react-native';
import React, { useState, useEffect } from 'react';

import { getDownloadURL, getStorage, ref } from 'firebase/storage';

export default function MenuItem(props) {
    const [itemUrl, setItemUrl] = useState('/Users/noahwalker/AVRS/AVRS_RNCLI/src/img/white.png');

    const getImage = (image, name) => {
        const storage = getStorage();
        console.log("getImage", image);
        const starsRef = ref(storage, image);
        let test = "";

        getDownloadURL(starsRef)
            .then((url) => {
                // console.log("2 works", url);

                // setSource([]);
                // let urlTemp = source;

                // if(urlTemp.length === 0) {
                //     urlTemp.pop();
                // }
                // urlTemp.pop();
                // urlTemp.push(url);
                setItemUrl(url)

                // setSource(url); 
                console.log("2 source", url);
                return url;
            })
            .catch((error) => {
                console.log("2 not work", error.message);
                console.log(error);
            });
            // Need to add a snack component that gets an image specific to the snack
    }

    useEffect(() => {
        getImage(props.thing.image, props.thing.Name);
    }, [])

    return (
        <View>
            <Pressable
                onPress={() => props.onSnackPress(props.thing.Name)}
                style={({ pressed }) => [
                    styles.menuItem,
                    pressed ? { opacity: 0.8 } : {},
                ]}>
                
                <Image
                    style={styles.imageStyle}
                    source={{
                        uri: itemUrl
                    }} 
                    />  
                <Text>{props.thing.Name}</Text>
            </Pressable>
            {/* <Text>{itemUrl}</Text> */}
        </View>
  )
}

const styles = StyleSheet.create({
    imageStyle: {
        width: 100,
        height: 100,
    },
	container: {
		flex: 1,
		justifyContent: 'center',
        alignItems: 'center',
        width: '100%',
	},
    headerContainer: {
        flex: 1,
        justifyContent: 'center',
        // backgroundColor: 'red',
        width: '100%',
    },
    snacksContainer: {
        flex: 5,
        width: '100%',
        // backgroundColor: 'white',
        // alignItems: 'center',
		justifyContent: 'center',
        flexDirection: 'row',
        flexWrap: 'wrap',
    },
	menuItem: {
        alignItems: 'center',
		justifyContent: 'center',
		width: 150,
        height: 150,
        margin: 10,
        backgroundColor: 'white',
        borderColor: 'black',
        borderWidth: 1,
        borderRadius: 10,
	},
    header: {
        margin: 10,
        fontSize: 30,
        fontWeight: 'bold',
    },
	navBar: {
		flexDirection: 'row',
		width: '90%',
		justifyContent: 'space-evenly',
	}

})