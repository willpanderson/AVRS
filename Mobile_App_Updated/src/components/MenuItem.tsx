import {View, Text, Pressable, Image, StyleSheet} from 'react-native';
import React, {useState, useEffect} from 'react';

import {getDownloadURL, ref} from 'firebase/storage';
import {SnackType} from '../@types';
import {storage} from '../../firebase';

export default function MenuItem(props: {
  snack: SnackType;
  onSnackPress: (arg0: any) => void;
}) {
  const [itemUrl, setItemUrl] = useState(
    '/Users/noahwalker/AVRS/AVRS_RNCLI/src/img/white.png',
  );

  const getImage = (image: string | undefined) => {
    const starsRef = ref(storage, image);

    getDownloadURL(starsRef)
      .then(url => {
        setItemUrl(url);
        return url;
      })
      .catch(error => {
        console.log('2 not work', error.message);
        console.log(error);
      });
  };

  useEffect(() => {
    getImage(props.snack.image);
  }, [props.snack.Name, props.snack.image]);

  return (
    <View>
      <Pressable
        onPress={() => props.onSnackPress(props.snack.Name)}
        style={({pressed}) => [styles.menuItem, pressed ? {opacity: 0.8} : {}]}>
        <Image
          style={styles.imageStyle}
          source={{
            uri: itemUrl,
          }}
        />
        <Text>{props.snack.Name}</Text>
      </Pressable>
    </View>
  );
}

const styles = StyleSheet.create({
  imageStyle: {
    width: 100,
    height: 100,
  },
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    width: '100%',
  },
  headerContainer: {
    flex: 1,
    justifyContent: 'center',
    width: '100%',
  },
  snacksContainer: {
    flex: 5,
    width: '100%',
    justifyContent: 'center',
    flexDirection: 'row',
    flexWrap: 'wrap',
  },
  menuItem: {
    alignItems: 'center',
    justifyContent: 'center',
    width: 150,
    height: 150,
    margin: 10,
    backgroundColor: 'white',
    borderColor: 'black',
    borderWidth: 1,
    borderRadius: 10,
  },
  header: {
    margin: 10,
    fontSize: 30,
    fontWeight: 'bold',
  },
  navBar: {
    flexDirection: 'row',
    width: '90%',
    justifyContent: 'space-evenly',
  },
});
