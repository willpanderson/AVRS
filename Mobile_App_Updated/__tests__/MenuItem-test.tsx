import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import MenuItem from '../src/components/MenuItem';
import {fireEvent, render} from '@testing-library/react-native';
// import * as firebase from 'firebase/storage';

describe('Tests for MenuItem component', () => {
  const snack = {
    Name: 'snack',
    Reserved: 2,
    Total: 2,
    image: 'file.png',
  };

  const onSnackPress = jest.fn((name: string) => {
    console.log(name);
  });

  jest.mock('firebase/storage', () => ({
    getStorage: jest.fn(),
    getDownloadURL: jest.fn(),
    ref: jest.fn(),
  }));

  it('Renders correctly', () => {
    const tree = renderer
      .create(<MenuItem snack={snack} onSnackPress={() => onSnackPress} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Click snack to add to cart', () => {
    // jest.spyOn(firebase, 'getDownloadURL').mockImplementation(() => {
    //   return new Promise(() => {
    //     return 'hello';
    //   });
    // });

    const {getByText} = render(
      <MenuItem snack={snack} onSnackPress={onSnackPress} />,
    );

    fireEvent.press(getByText(snack.Name));
    expect(onSnackPress).toBeCalled();
    expect(true).toBe(true);
  });
});
