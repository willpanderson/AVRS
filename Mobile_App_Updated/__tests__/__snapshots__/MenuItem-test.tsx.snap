// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for MenuItem component Renders correctly 1`] = `
<View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderColor": "black",
          "borderRadius": 10,
          "borderWidth": 1,
          "height": 150,
          "justifyContent": "center",
          "margin": 10,
          "width": 150,
        },
        Object {},
      ]
    }
  >
    <Image
      source={
        Object {
          "uri": "/Users/noahwalker/AVRS/AVRS_RNCLI/src/img/white.png",
        }
      }
      style={
        Object {
          "height": 100,
          "width": 100,
        }
      }
    />
    <Text>
      snack
    </Text>
  </View>
</View>
`;
